tests/python/pants_test/option/test_options.py

```diff
@@ -789,6 +803,24 @@ def test_drop_flag_values(self):
     self.assertEqual('red', options.for_global_scope().pants_foo)
     self.assertEqual('BAR', defaulted_only_options.for_global_scope().pants_foo)
 
+    # Overriding an enum option value.
+    self.assertEqual(self.SomeEnumOption.another_value, options.for_scope('enum-opt').some_enum)
+
+    # Getting the default value for an enum option.
+    self.assertEqual(
+      self.SomeEnumOption.a_value,
+      defaulted_only_options.for_scope('separate-enum-opt-scope').some_enum_with_default)
+
+  def test_enum_option_type_parse_error(self):
+    self.maxDiff = None
+    with self.assertRaises(ParseError) as cm:
+      options = self._parse('./pants enum-opt --some-enum=invalid-value')
+      options.for_scope('enum-opt').some_enum
+    self.assertIn("""\
+ParseError: Error applying type 'SomeEnumOption' to option value 'invalid-value', for option '--some_enum' in scope 'enum-opt': type check error in class SomeEnumOption: Value {u}'invalid-value' must be one of: [{u}'a-value', {u}'another-value']."""
```

Comment:
this is quite specific, maybe shave it down to the just the errors args such as `invalid-value` etc. any slight change to the exception message will make this fail.

Reply:
Fixed! This was proven correct as an incorrect error message here was causing the CI failure.

Reply:
looks good!